test/checks/content_for_header_modification_test.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+require "test_helper"
+
+class ContentForHeaderModificationTest < Minitest::Test
+  def test_reports_use_of_filter
+    offenses = analyze_theme(
+      ThemeCheck::ContentForHeaderModification.new,
+      "layout/theme.liquid" => <<~END,
+        {{ content_for_header | split: ',' }}
+      END
+    )
+    assert_offenses(<<~END, offenses)
+      Do not rely on the content of `content_for_header` at layout/theme.liquid:1
+    END
+  end
+
+  def test_reports_assign
+    offenses = analyze_theme(
+      ThemeCheck::ContentForHeaderModification.new,
+      "layout/theme.liquid" => <<~END,
+        {% assign x = content_for_header %}
+      END
+    )
+    assert_offenses(<<~END, offenses)
+      Do not rely on the content of `content_for_header` at layout/theme.liquid:1
+    END
+  end
+
+  def test_reports_capture
+    offenses = analyze_theme(
+      ThemeCheck::ContentForHeaderModification.new,
+      "layout/theme.liquid" => <<~END,
+        {% capture x %}
+          {{ content_for_header }}
+        {% endcapture %}
+      END
+    )
+    assert_offenses(<<~END, offenses)
+      Do not rely on the content of `content_for_header` at layout/theme.liquid:2
+    END
+  end
+
+  def test_do_not_report_normal_use
+    offenses = analyze_theme(
+      ThemeCheck::ContentForHeaderModification.new,
+      "layout/theme.liquid" => <<~END,
+        {{ content_for_header }}
+      END
+    )
+    assert_offenses("", offenses)
```

Comment:
Should we add a test case for `{% liquid %}` blocks and `echo`?

Reply:
Do you mean check for `echo content_for_header`?

Reply:
Yes just make sure that still works but not `echo content_for_header | ...`. Probably redundant but I've been bit in the past.